src/Orleans/Messaging/BufferPool.cs

```diff
@@ -82,6 +89,11 @@ public byte[] GetBuffer()
                 buffer = new byte[byteBufferSize];
                 allocatedBufferCounter.Increment();
             }
+            else if (limitBuffersCount)
+            {
+                Interlocked.Decrement(ref currentAllocatedBuffers);
```

Comment:
Used separate counter with Interlocked as it twice more faster that accessing of ConcurrentBag Count property. But the difference in global perspective isn't much noticeable, so in readability purposes can change to .Count.